checks/branch_protection.go

```diff
@@ -102,25 +97,42 @@ func checkReleaseAndDevBranchProtection(ctx context.Context, r repositories, dl
 	}
 	checkBranches[*repo.DefaultBranch] = true
 
-	// Check protections on the branches.
+	protected := true
+	// Check protections on all the branches.
 	for b := range checkBranches {
-		protected, err := isBranchProtected(branches, b)
+		p, err := isBranchProtected(branches, b)
 		if err != nil {
-			r := checker.CreateRuntimeErrorResult(CheckBranchProtection, err)
-			checks = append(checks, r)
+			return checker.CreateRuntimeErrorResult(CheckBranchProtection, err)
 		}
-		if !protected {
-			r := checker.CreateMinScoreResult(CheckBranchProtection,
-				fmt.Sprintf("branch protection not enabled for branch '%s'", b))
-			checks = append(checks, r)
+		if !p {
+			protected = false
+			dl.Warn("branch protection not enabled for branch '%s'", b)
 		} else {
 			// The branch is protected. Check the protection.
-			res := getProtectionAndCheck(ctx, r, dl, ownerStr, repoStr, b)
-			checks = append(checks, res)
+			score, err := getProtectionAndCheck(ctx, r, dl, ownerStr, repoStr, b)
+			if err != nil {
+				return checker.CreateRuntimeErrorResult(CheckBranchProtection, err)
+			}
+			scores = append(scores, score)
 		}
 	}
 
-	return checker.MakeAndResult2(checks...)
+	if !protected {
+		return checker.CreateMinScoreResult(CheckBranchProtection,
+			"branch protection not enabled on development/release branches")
```

Comment:
*all
(in other places too) just to clarify?

Reply:
by all, you mean it should say `all branches`? I thought the code only look for main and release branches.

Reply:
ah, mean "development and all release branches"
but not a big point, since logging probably covers it

Reply:
done.